KNMI/adaguc-server
# Optimize addfile script and clean based on reference_time for models for WIK

**Version 2.13.0 2023-09-20**
- The script adaguc-server-addfile.sh is now able to find for a given file the corresponding dataset via the commandline option autofinddataset
- Cleaning of model data is now done based on the forecast_reference_time dimension
- Logging can be made less verbose by adding --verboseoff

# How test:

- copy the file `data/datasets/testdata.nc` to the `/data/adaguc-data/ folder`
- Make a dataset config in the /data/adaguc-datasets folder like:

```xml
<?xml version="1.0" encoding="UTF-8" ?>
<Configuration>
  <Layer type="database">
    <FilePath filter="">/data/adaguc-data/testdata.nc</FilePath>
     <Variable>testdata</Variable>
    <Styles>auto</Styles>
  </Layer>
</Configuration>
``` 
- Make sure your adaguc environment is correctly setup
- run the command `bash ./Docker/adaguc-server-addfile.sh -f /data/adaguc-data/testdata.nc`

It should auto detect to which dataset this file belongs.
